restapi/src/main/java/io/stargate/web/models/IndexAdd.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.web.models;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.swagger.annotations.ApiModelProperty;
+import javax.validation.constraints.NotNull;
+
+public class IndexAdd {
+  @NotNull private String table;
+  @NotNull private String column;
+  private String name;
+  private String type;
+  private IndexKind kind;
+
+  private boolean ifNotExists = false;
+
+  public void setTable(String table) {
+    this.table = table;
+  }
+
+  @ApiModelProperty(required = true, value = "Table name")
+  public String getTable() {
+    return table;
+  }
+
+  public void setColumn(String column) {
+    this.column = column;
+  }
+
+  @ApiModelProperty(required = true, value = "Column name")
+  public String getColumn() {
+    return column;
+  }
+
+  public void setName(String name) {
+    this.name = name;
+  }
+
+  @ApiModelProperty(value = "The name of the index to add.")
```

Comment:
Would you mind adding something about what happens if the name isn't provided? In the future we can add a `getAllIndexesForTable` but for now it'd be good to let people know how to delete an index they didn't name.

Reply:
Sure! Gonna add that.

Reply:
Thanks!